### DISCUSSION TITLE
webpack interfering with elite frameworks - - Error: Module not found: Error: Can't resolve 'fs'

### DISCUSSION THREAD

**Discussion by nhhockeyplayer** [webpack/webpack#13699] [General]
SUBJECT: - Error: Module not found: Error: Can't resolve 'fs'

This error is rampant all over the web and intersects across many build methods 

WEBPACK is forcing our SPA build configuration to create an angular webpack builder and add a configuration to the webpack.config.js as part of our build in order to get past this as follows

```
"builder": "@angular-builders/custom-webpack:browser",
"options": {
"customWebpackConfig": {
"path": "./custom-webpack.config.js"
},
```

and the following code within webpack.config.js is needed to overcome this specifically the "target" property
```
module.exports = {
    target: "node",
...
```

None of this should be necessary
webpack is not specified in my monorepo which is characterized by nrwl nx angula 

My SPA does not even use webpack nor includes it as part of my package.json

webpack has been concealed by the angular team effectively already and mitigated for no configuration

webpack insists on producing this error 
it stops blocks our build process hampering development for much time while losing 95% away from focusing on pure business logic and all time spent in build hell

# Bug report

**What is the current behavior?**
Currently building with latest Nrwl/Nx and Angular; it breaks my build with this error

`./node_modules/@angular/compiler-cli/src/ngtsc/file_system/src/node_js_file_system.mjs:9:0-25 - Error: Module not found: Error: Can't resolve 'fs' in '/Users/me/app/node_modules/@angular/compiler-cli/src/ngtsc/file_system/src'`

**If the current behavior is a bug, please provide the steps to reproduce.**
My repo is internal and large unable to offer a repo

200k+ lines of code end to end apps and servers at least 7 projects within a Nrwl monorepo

**What is the expected behavior?**

our builds should complete without ANY attention to webpack by design

webpack is imposing and pushing its way in and its not asked for

inspired workarounds to this is as follows... added to package.json but this currently fails to mitigate this issue

package.json
```
    "browser": {
        "fs": "empty"
    }
```
but this is insufficient

our builds should complete gracefully at this juncture but they do not from the above error listed

mind you every build I do starts fresh with deleted lock files and removed node_modules CLEAN from the start fresh

**Other relevant information:**
webpack version:   5
Node.js version: 12
Operating System: macos
Additional tools: nrwl nx cli; angular cli, npm, yarn

**Comment by alexander-akait** [comment]
@nhhockeyplayer 

> If the current behavior is a bug, please provide the steps to reproduce.

unfortunately in this case it is difficult to say what the problem is

What is your target browser or node?In browser there is no `fs`

**Reply by My69goodtimes** [reply]
Ok

**Comment by My69goodtimes** [comment]
Ok then

**Reply by My69goodtimes** [reply]
Ware it at

**Comment by nhhockeyplayer** [comment]
webpack insists on being part of an angular 12 project

no way around it for me

this was my main complaint

until angular team gets time and smarts to figure what webpack AND NPM are failing on

one has to add webpack and config and add the right stuff as noted in other issues

**Reply by alexander-akait** [reply]
I don't understand you
